src/slices/AccountSlice.ts

```diff
@@ -286,34 +288,48 @@ export const getDonationBalances = createAsyncThunk(
 
       try {
         // NOTE: The BigNumber here is from ethers, and is a different implementation of BigNumber used in the rest of the frontend. For that reason, we convert to string in the interim.
+        const depositIds: BigNumber[] = await givingContract.getDepositorIds(address);
         const allDeposits: [string[], BigNumber[]] = await givingContract.getAllDeposits(address);
-        for (let i = 0; i < allDeposits[0].length; i++) {
-          if (allDeposits[1][i].eq(0)) continue;
-          const depositAmount = ethers.utils.formatUnits(allDeposits[1][i], "gwei");
-          const recipient = allDeposits[0][i];
-          const firstDonationDate: string = await GetDonationDate({
-            address: address,
-            recipient: recipient,
-            networkID: networkID,
-            provider: provider,
-          });
-
-          // NOTE: Bad fix, but since no rebases on testnet this would throw an error otherwise
-          const yieldSent: BigNumber =
-            networkID === 1 ? await givingContract.donatedTo(address, recipient) : BigNumber.from("0");
-          const formattedYieldSent = ethers.utils.formatUnits(yieldSent, "gwei");
-
-          donationInfo.push({
-            date: firstDonationDate,
-            deposit: depositAmount,
-            recipient: recipient,
-            yieldDonated: formattedYieldSent,
-          });
+
+        // ([0x00000...], [0]) is the null return indicating a user has no deposits
+        // This confirms that we did not receive back a null deposit
+        if (allDeposits[0].length != 1 || allDeposits[1][0] != BigNumber.from(0)) {
+          for (let i = 0; i < allDeposits[0].length; i++) {
+            if (allDeposits[1][i].eq(0)) continue;
+            const depositId = depositIds[i];
+            const sohmValue = await gohmContract.balanceFrom(allDeposits[1][i]);
+            const depositAmount = ethers.utils.formatUnits(sohmValue, "gwei");
+            const recipient = allDeposits[0][i];
+            const firstDonationDate: string = await GetDonationDate({
+              address: address,
+              recipient: recipient,
+              networkID: networkID,
+              provider: provider,
+            });
+
+            // NOTE: Bad fix, but since no rebases on testnet this would throw an error otherwise
+            const yieldSent: BigNumber =
+              networkID === 1
+                ? await getTotalYieldSent({
+                    address: address,
+                    recipient: recipient,
+                    networkID: networkID,
+                    provider: provider,
+                  })
+                : BigNumber.from("0");
+            const sohmYieldSent = await gohmContract.balanceFrom(yieldSent);
```

Comment:
There are two async calls here, within a for loop. As the number of deposits increases, isn’t this going to become a performance bottleneck? You may want to look at batching the async calls. If out of scope for this PR/sprint, then log a task.